autogpt_platform/backend/backend/util/service.py

```diff
@@ -99,16 +100,24 @@ def custom_dict_to_class(qualname, data: dict):
     return custom_dict_to_class
 
 
-class AppService(AppProcess):
+class AppService(AppProcess, ABC):
     shared_event_loop: asyncio.AbstractEventLoop
     use_db: bool = False
     use_redis: bool = False
     use_supabase: bool = False
 
-    def __init__(self, port):
-        self.port = port
+    def __init__(self):
         self.uri = None
 
+    @classmethod
+    @abstractmethod
+    def get_port(cls) -> int:
+        pass
+
+    @classmethod
+    def get_host(cls) -> str:
+        return os.environ.get(f"{cls.service_name.upper()}_HOST", Config().pyro_host)
```

Comment:
Is there a reason settings can’t be used here

Reply:
the pyro_host uses Settings,
For the dynamic config, idk, it's dynamic after all. I think @aarushik93 introduced this for prod purpose, e.g: dynamically separating host between services.